core/domain/prod_validation_jobs_one_off.py

```diff
@@ -15,28 +15,133 @@
+import itertools
+import re
+
+from constants import constants
 from core import jobs
+from core.domain import activity_domain
+from core.domain import collection_domain
+from core.domain import collection_services
+from core.domain import config_domain
+from core.domain import exp_domain
+from core.domain import exp_services
+from core.domain import fs_domain
+from core.domain import recommendations_services
+from core.domain import rights_manager
+from core.domain import story_domain
+from core.domain import story_services
 from core.platform import models
+import feconf
 
-(user_models, exp_models, collection_models, feedback_models) = (
-    models.Registry.import_models([
-        models.NAMES.user, models.NAMES.exploration,
-        models.NAMES.collection, models.NAMES.feedback]))
+(
+    activity_models, audit_models, base_models,
+    collection_models, config_models, email_models,
+    exp_models, feedback_models, file_models,
+    recommendations_models, story_models,
+    user_models,) = (
+        models.Registry.import_models([
+            models.NAMES.activity, models.NAMES.audit, models.NAMES.base_model,
+            models.NAMES.collection, models.NAMES.config, models.NAMES.email,
+            models.NAMES.exploration, models.NAMES.feedback, models.NAMES.file,
+            models.NAMES.recommendations, models.NAMES.story,
+            models.NAMES.user]))
 datastore_services = models.Registry.import_datastore_services()
 
+ALLOWED_AUDIO_EXTENSIONS = feconf.ACCEPTED_AUDIO_EXTENSIONS.keys()
+ALLOWED_IMAGE_EXTENSIONS = list(itertools.chain.from_iterable(
+    feconf.ACCEPTED_IMAGE_FORMATS_AND_EXTENSIONS.values()))
+ASSETS_PATH_REGEX = '/exploration/[A-Za-z0-9]{1,12}/assets/'
+IMAGE_PATH_REGEX = (
+    '%simage/[A-Za-z0-9_]{1,}\\.(%s)' % (
+        ASSETS_PATH_REGEX, ('|').join(ALLOWED_IMAGE_EXTENSIONS)))
+AUDIO_PATH_REGEX = (
+    '%saudio/[A-Za-z0-9_]{1,}\\.(%s)' % (
+        ASSETS_PATH_REGEX, ('|').join(ALLOWED_AUDIO_EXTENSIONS)))
+FILE_MODELS_REGEX = '(%s|%s)' % (IMAGE_PATH_REGEX, AUDIO_PATH_REGEX)
+
 
 class BaseModelValidator(object):
     """Base class for validating models."""
 
-    errors = {}
-    # external_models is keyed by field name. Each value consists
-    # of the model class and a list of (external_key, external_model_instance)
+    errors = collections.defaultdict(list)
+    # external_instance_details is keyed by field name. Each value consists
+    # of a list of (model class, external_key, external_model_instance)
     # tuples.
-    external_models = {}
+    external_instance_details = {}
+
+    @classmethod
+    def _get_model_id_regex(cls, unused_item):
+        """Returns a regex for model id.
+
+        This method can be overridden by subclasses, if needed.
+
+        Args:
+            unused_item: ndb.Model. Entity to validate.
+
+        Returns:
+            str. A regex pattern to be followed by the model id.
+        """
+        return '^[A-Za-z0-9]{1,%s}$' % base_models.ID_LENGTH
```

Comment:
Note: you will need to add "-" and "_" to the charset for ExplorationModel and CollectionModel (and models that were around prior to 2018 or so that used random ID construction).

Reply:
Done.

Reply:
Do you mean this for all models that were introduced in the codebase prior to 2018?